tools/android/android_sdk_downloader/lib/src/zip.dart

```diff
@@ -0,0 +1,41 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+
+// todo: if/when a streaming unzip routine is available for Dart, use that instead.
+
+Future<void> unzipFile(String file, Directory outDir) async {
+  await outDir.parent.create(recursive: true);
+  final Directory tempDir = await outDir.parent.createTemp();
+
+  String command;
+  List<String> args;
+  if (Platform.isWindows) {
+    command = 'powershell.exe  -nologo -noprofile -command '
+        '"& { '
+        'Add-Type -A \'System.IO.Compression.FileSystem\'; '
+        '[IO.Compression.ZipFile]::ExtractToDirectory(\'$file\', \'${tempDir.path}\'); '
```

Comment:
note: this mechanism of extracting things is kinda super-slow. I know, it's also the only one that available on pretty much all Windows platforms, so...